web/src/navigator.rs

```diff
@@ -33,6 +40,17 @@ impl WebNavigatorBackend {
             performance,
             allow_script_access,
             upgrade_to_https,
+            base_url,
+        }
+    }
+
+    fn base_uri(&self, document: &Document) -> Option<String> {
```

Comment:
Ideally this should probably return a `Cow`, to avoid copying the base_url... but that doesn't really matter much here.